core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -60,6 +61,7 @@ final class TableScanContext {
     this.planExecutor = null;
     this.fromSnapshotInclusive = false;
     this.scanReporter = new LoggingScanReporter();
+    this.ref = null;
```

Comment:
Technically with the current scan support we don't even need to store ref in the context. at the end of the day with the ref scanning we're scanning from a single snapshot, so we could just resolve the snapshot and set it as before. I'll think some more if this is really needed.